app/view/pages/github/index.js

```diff
@@ -0,0 +1,95 @@
+import React, {useEffect, useState} from 'react';
+import {Animated} from 'react-native';
+import {useSelector, useDispatch} from 'react-redux';
+import {useNavigation} from 'react-navigation-hooks';
+
+import Item from './Item';
+import {View, FlatList, Loading, Appbar, Button, Text} from 'view/components';
+
+import i18n from 'i18n';
+import style from 'view/style';
+
+import {getRepos} from 'store/reducers/github';
+import {selectGithub} from 'store/selectors/github';
+
+import buttons from './buttons';
+
+const Github = () => {
+  const {appLang, githubLang} = i18n;
+  const {appStyle, githubStyle} = style;
+
+  const [query, setQuery] = useState('react');
+  const [scaleValue] = useState(new Animated.Value(0));
+
+  const dispatch = useDispatch();
+  const {goBack} = useNavigation();
+  const github = useSelector(selectGithub);
+
+  const _goBack = () => goBack();
+
+  useEffect(() => {
+    dispatch(getRepos(query));
+  }, [dispatch, query]);
+
+  useEffect(() => {
+    if (!github.loading && github.data.length) {
+      Animated.timing(scaleValue, {
+        delay: github.data[query] * 350,
+        duration: 600,
+        toValue: 1,
+      }).start();
+    }
+  }, [github, query, scaleValue]);
+
+  return (
+    <View style={appStyle.container}>
+      <Appbar.Header style={appStyle.header}>
+        <Appbar.Action onPress={() => _goBack()} icon="arrow-back" />
+        <Appbar.Content title={githubLang.title} />
+      </Appbar.Header>
+
+      <View style={appStyle.content}>
+        <View style={githubStyle.buttonArea}>
+          {buttons.map(button => (
+            <Button
+              key={button.id}
+              mode="outlined"
+              color={'#13a77f'}
+              disabled={github.loading}
+              onPress={() => setQuery(button.id)}
+              style={appStyle.button}>
+              {github.loading
+                ? query === button.id
+                  ? appLang.loading
+                  : button.title
+                : button.title}
+            </Button>
+          ))}
+        </View>
+
+        <View style={githubStyle.flatList}>
+          {github.error ? (
```

Comment:
Actually, using ternary inside another ternary is bad for readability and loading components mean it could be a separated or more component. If I want to write this part, definitely, I decouple this part from its parent.

Reply:
I try to keep it inside because it's not used anywhere else, hope it's accepted

Reply:
Decoupling is not just for multiple usages. Maybe it could be decoupled because it does something independently special from its parent. In this special case, I preferred it could be decoupled because of good readability.

By the way, as you wish dude.

Reply:
I think it's ok by now cause I don't want to add another small component, anyway I wrapped strings in a text